libs/checkpoint/langgraph/checkpoint/serde/jsonplus.py

```diff
@@ -58,14 +58,14 @@ def _default(self, obj):
             return self._encode_constructor_args(
                 obj.__class__, method=[None, "construct"], kwargs=obj.dict()
             )
+        elif hasattr(obj, "_asdict") and callable(obj._asdict):
+            return self._encode_constructor_args(obj.__class__, kwargs=obj._asdict())
         elif isinstance(obj, pathlib.Path):
             return self._encode_constructor_args(pathlib.Path, args=obj.parts)
         elif isinstance(obj, re.Pattern):
             return self._encode_constructor_args(
                 re.compile, args=[obj.pattern, obj.flags]
             )
-        elif isinstance(obj, UUID):
```

Comment:
Why not just `str()`?

Reply:
Because then when we get it back it's not a UUID anymore, but will actually change, as that's the only option with orjson